File: src/Portfolio/Portfolio.Blazor/Data/Entities/WritingTranslation.cs
using Portfolio.Blazor.Data;

namespace Portfolio.Blazor.Data.Entities;

public sealed class WritingTranslation
{
    public int Id { get; set; }
    public int WritingId { get; set; }
    public string Locale { get; set; } = string.Empty;
    public string Title { get; set; } = string.Empty;
    public string? Excerpt { get; set; }
    public string? ReadingTime { get; set; }
    public string? Body { get; set; }

    [JsonOrEmpty]
    public string? Seo { get; set; }

    public DateTime? CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }

    public Writing? Writing { get; set; }
}

File: laravel/app/Models/WritingTranslation.php
<?php

namespace App\Models;

use App\Models\Concerns\Auditable;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class WritingTranslation extends Model
{
    use Auditable, HasFactory;

    protected $fillable = ['writing_id', 'locale', 'title', 'excerpt', 'reading_time', 'body', 'seo'];

    protected $casts = ['seo' => 'array'];

    /**
     * @return BelongsTo<Writing, $this>
     */
    public function writing(): BelongsTo
    {
        return $this->belongsTo(Writing::class);
    }
}
